lib/tapioca/dsl/compilers/active_record_typed_store.rb

```diff
@@ -95,11 +95,20 @@ class ActiveRecordTypedStore < Compiler
         sig { override.void }
         def decorate
           stores = constant.typed_stores
-          return if stores.values.flat_map(&:accessors).empty?
+          accessors = stores.values.flat_map(&:accessors)
+          return if accessors.empty?
+
+          # Support versions > v1.5.0
+          if accessors.first.is_a?(Hash)
+            return if accessors.filter_map { |accessor| accessor.keys.first }.empty?
+          end
```

Comment:
I am a little lost as to what is happening here. I think it makes sense to encapsulate and document some of this logic.

Reply:
`accessors` now returns an [array of hashes](https://github.com/byroot/activerecord-typedstore/pull/91/files#diff-64307a9bb8926d5501848c78723c0e10334b7a6d04b09897d11ae3647246152eR37-R38).

I'm returning early if any of those hashes are empty by checking if they have any keys. Were you thinking of adding a comment and extract it to a method?

Reply:
I find it nicer to unify APIs when possible. We could either turn a non-hash accessor to the new hashed one to unify on the new API or map the values of the hash accessor to a simple list to unify on the old API.

The resulting code ends up much cleaner that way.

Reply:
`flat_map(&:accessors)` returns `[{}]` if there's a store with no accessors, where previously you'd get an empty Array. I changed this bit to `all? { |store| store.accessors.empty? }` which works both with accessors being arrays or hashes.